src/System.Text.Primitives/System/Text/InvariantParser_ulong.cs

```diff
@@ -30,20 +30,23 @@ public static bool TryParse(Utf8String utf8Text, out ulong value, out int bytesC
                         return true;
                     }
                 }
-                try
+                else if (value > UInt64.MaxValue / 10) // overflow
                 {
-                    ulong candidate = checked(value * 10 + nextByte - '0');
-                    Debug.Assert(candidate >= value);
-
-                    value = candidate;
-                    bytesConsumed++;
+                    value = 0;
+                    bytesConsumed = 0;
+                    return false;
                 }
-                catch (OverflowException e)
+                else if (UInt64.MaxValue - value * 10 < (ulong)(nextByte - '0')) // overflow
                 {
                     value = 0;
                     bytesConsumed = 0;
                     return false;
                 }
+                ulong candidate = checked(value * 10 + nextByte - '0');
```

Comment:
Ah, good catch. We don't use the Utf8String class yet, so I hadn't made the change fully here.

I removed the asserts from the signed classes because we want to allow overflow in the case of the min value; but I may add it back if it proves useful in some cases.